// react-native-face-detection/lib/module/types.js
export let FaceDetectorClassificationMode;

(function (FaceDetectorClassificationMode) {
  FaceDetectorClassificationMode[FaceDetectorClassificationMode["NONE"] = 1] = "NONE";
  FaceDetectorClassificationMode[FaceDetectorClassificationMode["ALL"] = 2] = "ALL";
})(FaceDetectorClassificationMode || (FaceDetectorClassificationMode = {}));

export let FaceDetectorContourMode;

(function (FaceDetectorContourMode) {
  FaceDetectorContourMode[FaceDetectorContourMode["NONE"] = 1] = "NONE";
  FaceDetectorContourMode[FaceDetectorContourMode["ALL"] = 2] = "ALL";
})(FaceDetectorContourMode || (FaceDetectorContourMode = {}));

export let FaceDetectorLandmarkMode;

(function (FaceDetectorLandmarkMode) {
  FaceDetectorLandmarkMode[FaceDetectorLandmarkMode["NO"] = 1] = "NO";
  FaceDetectorLandmarkMode[FaceDetectorLandmarkMode["ALL"] = 2] = "ALL";
})(FaceDetectorLandmarkMode || (FaceDetectorLandmarkMode = {}));

export let FaceDetectorPerformanceMode;

(function (FaceDetectorPerformanceMode) {
  FaceDetectorPerformanceMode[FaceDetectorPerformanceMode["FAST"] = 1] = "FAST";
  FaceDetectorPerformanceMode[FaceDetectorPerformanceMode["ACCURATE"] = 2] = "ACCURATE";
})(FaceDetectorPerformanceMode || (FaceDetectorPerformanceMode = {}));

export let FaceContourType;

(function (FaceContourType) {
  FaceContourType[FaceContourType["FACE"] = 1] = "FACE";
  FaceContourType[FaceContourType["LEFT_EYEBROW_TOP"] = 2] = "LEFT_EYEBROW_TOP";
  FaceContourType[FaceContourType["LEFT_EYEBROW_BOTTOM"] = 3] = "LEFT_EYEBROW_BOTTOM";
  FaceContourType[FaceContourType["RIGHT_EYEBROW_TOP"] = 4] = "RIGHT_EYEBROW_TOP";
  FaceContourType[FaceContourType["RIGHT_EYEBROW_BOTTOM"] = 5] = "RIGHT_EYEBROW_BOTTOM";
  FaceContourType[FaceContourType["LEFT_EYE"] = 6] = "LEFT_EYE";
  FaceContourType[FaceContourType["RIGHT_EYE"] = 7] = "RIGHT_EYE";
  FaceContourType[FaceContourType["UPPER_LIP_TOP"] = 8] = "UPPER_LIP_TOP";
  FaceContourType[FaceContourType["UPPER_LIP_BOTTOM"] = 9] = "UPPER_LIP_BOTTOM";
  FaceContourType[FaceContourType["LOWER_LIP_TOP"] = 10] = "LOWER_LIP_TOP";
  FaceContourType[FaceContourType["LOWER_LIP_BOTTOM"] = 11] = "LOWER_LIP_BOTTOM";
  FaceContourType[FaceContourType["NOSE_BRIDGE"] = 12] = "NOSE_BRIDGE";
  FaceContourType[FaceContourType["NOSE_BOTTOM"] = 13] = "NOSE_BOTTOM";
  FaceContourType[FaceContourType["LEFT_CHECK"] = 14] = "LEFT_CHECK";
  FaceContourType[FaceContourType["RIGHT_CHECK"] = 15] = "RIGHT_CHECK";
})(FaceContourType || (FaceContourType = {}));

export let FaceLandmarkType;

(function (FaceLandmarkType) {
  FaceLandmarkType[FaceLandmarkType["MOUTH_BOTTOM"] = 1] = "MOUTH_BOTTOM";
  FaceLandmarkType[FaceLandmarkType["MOUTH_RIGHT"] = 2] = "MOUTH_RIGHT";
  FaceLandmarkType[FaceLandmarkType["MOUTH_LEFT"] = 3] = "MOUTH_LEFT";
  FaceLandmarkType[FaceLandmarkType["LEFT_EAR"] = 4] = "LEFT_EAR";
  FaceLandmarkType[FaceLandmarkType["RIGHT_EAR"] = 5] = "RIGHT_EAR";
  FaceLandmarkType[FaceLandmarkType["LEFT_EYE"] = 6] = "LEFT_EYE";
  FaceLandmarkType[FaceLandmarkType["RIGHT_EYE"] = 7] = "RIGHT_EYE";
  FaceLandmarkType[FaceLandmarkType["LEFT_CHEEK"] = 8] = "LEFT_CHEEK";
  FaceLandmarkType[FaceLandmarkType["RIGHT_CHEEK"] = 9] = "RIGHT_CHEEK";
  FaceLandmarkType[FaceLandmarkType["NOSE_BASE"] = 10] = "NOSE_BASE";
})(FaceLandmarkType || (FaceLandmarkType = {}));
