// react-native-face-detection/lib/commonjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "FaceDetectorClassificationMode", {
  enumerable: true,
  get: function () {
    return _types.FaceDetectorClassificationMode;
  }
});
Object.defineProperty(exports, "FaceDetectorContourMode", {
  enumerable: true,
  get: function () {
    return _types.FaceDetectorContourMode;
  }
});
Object.defineProperty(exports, "FaceDetectorLandmarkMode", {
  enumerable: true,
  get: function () {
    return _types.FaceDetectorLandmarkMode;
  }
});
Object.defineProperty(exports, "FaceDetectorPerformanceMode", {
  enumerable: true,
  get: function () {
    return _types.FaceDetectorPerformanceMode;
  }
});
Object.defineProperty(exports, "FaceContourType", {
  enumerable: true,
  get: function () {
    return _types.FaceContourType;
  }
});
Object.defineProperty(exports, "FaceLandmarkType", {
  enumerable: true,
  get: function () {
    return _types.FaceLandmarkType;
  }
});
Object.defineProperty(exports, "FaceDetectorOptionsType", {
  enumerable: true,
  get: function () {
    return _types.FaceDetectorOptionsType;
  }
});
Object.defineProperty(exports, "FaceResult", {
  enumerable: true,
  get: function () {
    return _types.FaceResult;
  }
});
exports.default = void 0;

var _reactNative = require("react-native");

var _types = require("./types");

function createFaceDetectorOptions(options) {
  const out = {
    classificationMode: _types.FaceDetectorClassificationMode.NONE,
    contourMode: _types.FaceDetectorContourMode.NONE,
    landmarkMode: _types.FaceDetectorLandmarkMode.NO,
    minFaceSize: 0.1,
    performanceMode: _types.FaceDetectorPerformanceMode.FAST
  };

  if (options === undefined || options === null) {
    return out;
  }

  if (options.classificationMode) {
    if (options.classificationMode !== _types.FaceDetectorClassificationMode.NONE && options.classificationMode !== _types.FaceDetectorClassificationMode.ALL) {
      throw new Error("'options.classificationMode' invalid classification mode. Expected FaceDetectorClassificationMode.NONE or FaceDetectorClassificationMode.ALL.");
    }

    out.classificationMode = options.classificationMode;
  }

  if (options.contourMode) {
    if (options.contourMode !== _types.FaceDetectorContourMode.NONE && options.contourMode !== _types.FaceDetectorContourMode.ALL) {
      throw new Error("'options.contourMode' invalid contour mode. Expected FaceDetectorContourMode.NONE or FaceDetectorContourMode.ALL.");
    }

    out.contourMode = options.contourMode;
  }

  if (options.landmarkMode) {
    if (options.landmarkMode !== _types.FaceDetectorLandmarkMode.NO && options.landmarkMode !== _types.FaceDetectorLandmarkMode.ALL) {
      throw new Error("'options.landmarkMode' invalid landmark mode. Expected FaceDetectorLandmarkMode.NO or FaceDetectorLandmarkMode.ALL.");
    }

    out.landmarkMode = options.landmarkMode;
  }

  if (options.minFaceSize !== null && options.minFaceSize !== undefined) {
    if (isNaN(options.minFaceSize)) {
      throw new Error("'options.minFaceSize' expected a number value between 0 & 1.");
    }

    if (options.minFaceSize < 0 || options.minFaceSize > 1) {
      throw new Error("'options.minFaceSize' expected value to be between 0 & 1.");
    }

    out.minFaceSize = options.minFaceSize;
  }

  if (options.performanceMode) {
    if (options.performanceMode !== _types.FaceDetectorPerformanceMode.FAST && options.performanceMode !== _types.FaceDetectorPerformanceMode.ACCURATE) {
      throw new Error("'options.performanceMode' invalid performance mode. Expected FaceDetectorPerformanceMode.FAST or FaceDetectorPerformanceMode.ACCURATE.");
    }

    out.performanceMode = options.performanceMode;
  }

  return out;
}

async function processImage(localImageFilePath, faceDetectorOption) {
  if (!localImageFilePath) {
    throw new Error("FaceDetection.processImage(*) 'localImageFilePath' expected a string local file path.");
  }

  const options = createFaceDetectorOptions(faceDetectorOption);
  const faces = await _reactNative.NativeModules.FaceDetection.processImage(localImageFilePath, options);
  return faces;
}

var _default = {
  processImage
};
exports.default = _default;
